core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java

```diff
@@ -437,8 +471,19 @@ private void removeWorkflowIndex(Workflow workflow, boolean archiveWorkflow)
                                 workflow.getWorkflowId(), workflow.getStatus()));
             }
         } else {
-            // Not archiving, also remove workflow from index
-            indexDAO.asyncRemoveWorkflow(workflow.getWorkflowId());
+
+            // if the workflow is indexed because of communication failure between conductor and
+            // fusion, it has
+            // to be removed from es
+            if (isPollProcessing) {
+                LOGGER.debug(
+                        "removing from index workflow id{} archiveworkflow {} isPollProcessing {} ",
```

Comment:
"remove index for workflow. Id {}, archiveWorkflow {}"
Don't think we need isPollProcessing in this log. Please check case for beginning of debug message.